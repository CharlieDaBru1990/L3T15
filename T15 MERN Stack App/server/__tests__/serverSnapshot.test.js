const { app, serverInstance } = require("../server");
const getPort = require("get-port");
const mongoose = require("mongoose");

let port;

beforeAll(async () => {
  port = await getPort();

  // Check if the server is already listening
  if (!serverInstance.listening) {
    serverInstance.listen(port);
    return new Promise((resolve) => {
      serverInstance.on("listening", () => {
        resolve();
      });
    });
  }
});

afterEach(async () => {
  await serverInstance.close();
  await mongoose.connection.close();
});

test("should match snapshot", () => {
  // Use snapshot matchers on the relevant part of the app object
  const appSnapshot = {
    routes: app._router.stack
      .filter((layer) => layer.route)
      .map((layer) => layer.route.path),
  };

  // If there are no routes, exclude the "routes" property from the snapshot
  if (appSnapshot.routes.length === 0) {
    delete appSnapshot.routes;
  }

  expect(appSnapshot).toMatchSnapshot();
});
