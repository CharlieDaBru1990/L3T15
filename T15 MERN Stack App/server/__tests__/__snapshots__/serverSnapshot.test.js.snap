// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server Snapshot should match snapshot 1`] = `{}`;

exports[`should match snapshot 1`] = `{}`;
